std/uni.d

```diff
@@ -8214,6 +8215,162 @@ unittest //12428
     assert(s == "abcdefghij");
 }
 
+
+// generic toUpper/toLower on whole range, returns range
+private auto toCaser(alias indexFn, uint maxIdx, alias tableFn, Range)(Range str)
+    if (isInputRange!Range && isSomeChar!(ElementEncodingType!Range))
+{
+    static if (!is(Unqual!(ElementEncodingType!Range) : const(dchar)))
```

Comment:
> I personally don't think the discrepancy is worth it.

Having spent considerable effort performance tuning this exact issue, it is definitely worth it.

As for the replacementChar thing, I discovered while writing byDchar that is the preferred method of web developers over throwing exceptions, because user input is often malformed and refusing to display the page because of that is a bad solution.

Having components always succeed is a good usability win.

I'd like to see all the autodecoding uses in Phobos replaced.